internal/templates/testing.tmpl

```diff
@@ -26,27 +26,17 @@ type Testing{{.BackendsName}} struct {
 {{end}}{{end}}
 }
 
-{{range .Deps}}
+{{range .Deps}}{{if not .IsDuplicate }}
 func (ti *Testing{{$.BackendsName}}) {{.Getter}}() {{.Type}} {
     if ti.{{.Var}} != nil {
        return ti.{{.Var}}
     }
 
-	return ti.Backends.{{.Getter}}()
+	return ti.{{$.BackendsName}}.{{.Getter}}()
 }
-{{end}}
-
-{{if .TransBcks -}}
-// Transitive dependency interface assertions.
-var (
-{{range .TransBcks -}}
-	_ {{.}} = (*Testing{{$.BackendsName}})(nil)
-{{end}}
-)
-{{end}}
-
+{{end}}{{end}}
 
-{{range .Deps }}
+{{range .Deps }}{{if not .IsDuplicate }}
 // Set{{.Getter}}ForTesting is a runtime available override for the "{{.Var}}" dependency that should only be used for testing.
 func (ti *Testing{{$.BackendsName}}) Set{{.Getter}}ForTesting(t *testing.T, {{.Var}} {{.Type}}) {
     t.Cleanup(func() {
```

Comment:
> `t.Cleanup(func() { ti.{{.Var}} = bil })`

Is setting it to nil the right thing to do? I also worry about tests clobbering each other if they're reusing a Backends. Either way, I think it's safer to store the current value and then reset to the current value, rather than nil.

Reply:
This was ultimately something that was discussed in another MR. I originally proposed using the original type and therefore setting and resetting but there wasnt any support for that so it currently wraps the Backends and has its own set of fields that are non-nil when overridden and nil when not.

Reply:
See here: https://github.com/luno/weld/pull/3